### PR TITLE
Fix reload and retest buttons in submissions details page on different resolution

### PR DESCRIPTION
**Pull request by PetarNeshkov**
Closes: https://github.com/SoftUni-Internal/exam-systems-issues/issues/520

### CODE REVIEW

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx [hunk 1]
```diff
@@ -12,40 +8,12 @@ const RefreshableSubmissionsList = ({
...
-        </>
-    );
-};
+}: IRefreshableSubmissionsListProps) => (
+    <SubmissionsList
+      items={items}
+      selectedSubmission={selectedSubmission}
+      className={className}
+    />
+);
```
Comment (mishogv) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx:17]: The refreshable list should handle Reloading logic internal and only expose callbacks or events where different implementations will be able to add

**Review by mishogv — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx [hunk 2]
```diff
@@ -146,8 +170,17 @@ const SubmissionDetails = () => {
         },
         [ canAccessAdministration, getAdministrationRetestSubmissionInternalUrl ],
     );
+    const buttonsSection = useMemo(
+        () => (
+            <div className={styles.buttonsSection}>
+                { renderReloadButton }
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx:176]: these functions are not called

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx [hunk 3]
```diff
@@ -198,11 +231,11 @@ const SubmissionDetails = () => {
                   selectedSubmission={currentSubmission}
                   className={styles.submissionsList}
                 />
-                { renderRetestButton() }
-                { renderSubmissionInfo() }
+                { buttonsSection }
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx:234]: these too

**Review by shsimeonova — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx [hunk 4]
```diff
@@ -1,51 +1,24 @@
...
-
-import { useSubmissionsDetails } from '../../../hooks/submissions/use-submissions-details';
-import Button, { ButtonType } from '../../guidelines/buttons/Button';
+import React from 'react';
 
 import SubmissionsList, { ISubmissionsListProps } from './SubmissionsList';
 
-type IRefreshableSubmissionsListProps = ISubmissionsListProps
+interface IRefreshableSubmissionsListProps extends ISubmissionsListProps{
+    itemFunc?: () => React.ReactElement;
```
Comment (mishogv) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx:6]: what means itemFunc

**Review by mishogv — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx [hunk 5]
```diff
@@ -146,8 +170,17 @@ const SubmissionDetails = () => {
         },
         [ canAccessAdministration, getAdministrationRetestSubmissionInternalUrl ],
     );
+    const buttonsSection = useCallback(
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx:173]: empty row here

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx [hunk 6]
```diff
@@ -1,51 +1,24 @@
...
-
+    itemFunc,
+}: IRefreshableSubmissionsListProps) => (
+    <>
+        <SubmissionsList
+          items={items}
+          selectedSubmission={selectedSubmission}
+          className={className}
+        />
+        {itemFunc?.()}
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx:21]: What does this function do?

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx [hunk 7]
```diff
@@ -128,7 +128,31 @@ const SubmissionDetails = () => {
...
+    const submissionsReloadBtnClassName = 'submissionReloadBtn';
+
+    const handleReloadClick = useCallback(async () => {
+        if (isNil(currentSubmission)) {
+            return;
+        }
+
+        const { problem: { id: problemId }, isOfficial, user: { id: userId } } = currentSubmission;
+
+        await getSubmissionResults(problemId, isOfficial, userId);
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx:140]: Why is the userId needed to get the submission results?

**Review by shsimeonova — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx (hunk 4 above)
Comment (PetarNeshkov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx:6]: We use itemFunc to pass JSX elements externally. I changed the name to be more understandable.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx (hunk 6 above)
Comment (PetarNeshkov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx:21]: We use itemFunc to pass JSX elements externally. I changed the name to be more understandable.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx [hunk 8]
```diff
@@ -1,51 +1,24 @@
...
-              onClick={handleReloadClick}
-              text="Reload"
-              type={ButtonType.secondary}
-              className={submissionsReloadBtnClassName}
-            />
-        </>
-    );
-};
-
+    externalElements,
```
Comment (mishogv) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx:13]: Add default function to return base button
functionName should be a verb
example name - renderExternalElements
```

            <Button
              onClick={handleReloadClick}
              text="Reload"
              type={ButtonType.secondary}
              className={submissionsReloadBtnClassName}
            />
```

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx [hunk 9]
```diff
@@ -111,7 +111,31 @@ const SubmissionDetails = () => {
...
+              onClick={handleReloadClick}
+              text="Reload"
+              type={ButtonType.secondary}
+              className={submissionsReloadBtnClassName}
+            />
+        ),
+        [ handleReloadClick ],
+    );
+
+    const renderRetestButton = useMemo(
```
Comment (mishogv) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx:138]: useMemo is only used for variables 
use useCallback instead

**Review by mishogv — CHANGES_REQUESTED** [review]

**Review by mishogv — APPROVED** [review]

**Comment by shsimeonova** [comment]
IMO the button for refreshing the list should be put outside of the component and only a refreshList function should be exposed, which will give us more flexibility on the styling and usage of the refresh button. Passing renderExternalElements to render the buttons section in the refreshable list breaks the single responsibility of the component and is not a good practice.

**Review by shsimeonova — DISMISSED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx
Comment (mishogv) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submissions-list/RefreshableSubmissionsList.tsx:1]: Delete the refreshable list component

**Review by mishogv — CHANGES_REQUESTED** [review]

**Review by mishogv — APPROVED** [review]

**Review by msmner — APPROVED** [review]

**Pull request merged**